src/core/blockchain/ethereum/client.ts

```diff
@@ -13,11 +13,13 @@ import { config } from './config';
 import { convertUnit } from '../ethereum/account';
 import abi from 'ethereumjs-abi';
 import { Erc20Client } from './tokens/erc20-client';
-import { TokenType } from '../types/token';
+import { TokenType, TokenScreenComponentType } from '../types/token';
 import { NameService } from './name-service';
 import { getTransactionStatusByCode } from './transaction';
 import { generateAccountTokenState } from '../../../redux/tokens/static-selectors';
 
+const inputRegex = /^(\w{10})(\w{24})(\w{40})(\w{64})$/;
```

Comment:
instead of regex, use `decodeInputData` function:
```ts
const decodeInputData = (signature, data) => {
  const sig = signature.split(":");
  const sigParts = sig[0].split("(");
  const methodName = sigParts[0];
  let params = [];
  if (sigParts[1]) {
    params = sigParts[1]
      .replace(")", "")
      .split(",")
      .map(p => p.trim());
  }

  const methodId = abi.methodID(methodName, params).toString("hex");

  data = data.replace("0x", "");

  if (data.indexOf(methodId) === 0) {
    rawParamsData = data.substr(methodId.length);
    return abi.rawDecode(params, Buffer.from(rawParamsData, "hex"));
  } else {
    throw new Error("Cannot decode data");
  }
};
```

Reply:
done